src/server/controllers/VerificationController-util.ts

```diff
@@ -71,6 +86,18 @@ export function isVerifiable(contractWrapper: ContractWrapper) {
         && Boolean(contractWrapper.chainId);
 }
 
+// TODO: Check if we already have the full file path or just the file name.
```

Comment:
Reverted to using your way of looking up and falling back to the first item otherwise.

What do you mean here with the TODO?

Reply:
I meant: "I have to learn if there is a better way to get the file name or ask to etherscan / sol devs if there is right way to extract the information"